services/src/main/java/org/sdo/demo/AioApiServlet.java

```diff
@@ -116,6 +118,43 @@ protected String getPath(String path) {
     return result;
   }
 
+  /**
+   *  Basic Path Traversal check.
+   */
+  public static boolean pathTraversalCheck(Path path) {
+    String homeDir = new AioDb().getProperty("fs.root.dir");
+    if (!(path.toString().contains("..")) && path.normalize().startsWith(homeDir)) {
```

Comment:
FYI - Checking for ".." is not enough. An attacker can substitute '..' to '%2e%2e'. Also, starts with defense is not valid as well. The best defense here would be to canonicalize the path and verify with startsWith.

Reply:
normalize() returns the canonicalized/absolute path. 

https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html#normalize%28%29
https://stackoverflow.com/questions/21552901/getting-canonical-path-from-path-in-java